src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs

```diff
@@ -1225,7 +1225,11 @@ public new CSharpCommandLineArguments Parse(IEnumerable<string> args, string bas
 
             // Dev11 searches for the key file in the current directory and assembly output directory.
             // We always look to base directory and then examine the search paths.
-            keyFileSearchPaths.Add(baseDirectory);
+            if (!string.IsNullOrWhiteSpace(baseDirectory))
+            {
+                keyFileSearchPaths.Add(baseDirectory);
+            }
+
             if (baseDirectory != outputDirectory)
```

Comment:
Note that if baseDirectory is non-null, but outputDirectory _is_ null, we'll end up with a null path in there again.

Reply:
It's difficult to construct this situation for testing because including a null in the `args` array causes NRE in the FlattenArgs method. Should I just modify like the following?

```cs
 if (!string.IsNullOrWhiteSpace(baseDirectory))
{
    keyFileSearchPaths.Add(baseDirectory);
    if (baseDirectory != outputDirectory)
    {
        keyFileSearchPaths.Add(outputDirectory);
    }
}
```

Reply:
Probably will just null check the outputDirectory before adding as we should add it even if the baseDirectory is null :)